src/java.base/share/classes/com/sun/crypto/provider/GaloisCounterMode.java

```diff
@@ -657,6 +697,72 @@ int getBufferedLength() {
             return (ibuffer == null ? 0 : ibuffer.size());
         }
 
+        /**
+         *  ByteBuffer wrapper for intrinsic implGCMCrypt
+         */
+        int implGCMCrypt(GCMOperation op, ByteBuffer src, ByteBuffer dst) {
+            int srcLen = src.remaining() - (src.remaining() % blockSize);
+            int rlen = srcLen;
+
+
+            if (srcLen < blockSize) {
+                return 0;
+            }
+
+            // 'in' and 'out' are always set together, just need to check 'in'
+            if (in == null || (in.length != PARALLEL_LEN
+                && in.length < srcLen)) {
+                in = new byte[Math.min(PARALLEL_LEN, srcLen)];
+                out = new byte[Math.min(PARALLEL_LEN, srcLen)];
+            }
+
+            byte[] ct;
+            if (encryption) {
+                ct = out;
+            } else {
+                ct = in;
+            }
```

Comment:
This can just be:
byte[] ct = (encryption? out : in);

Since you only use this 'ct' variable inside the else block on line  746, move this down to that block.